lib/usb/usb_standard.c

```diff
@@ -337,13 +375,26 @@ static int usb_standard_get_interface(usbd_device *usbd_dev,
 				      uint8_t **buf, uint16_t *len)
 {
 	uint8_t *cur_altsetting;
-	const struct usb_config_descriptor *cfx = &usbd_dev->config[usbd_dev->current_config - 1];
+	const struct usb_config_descriptor *cfx;
+
+	if (! usbd_dev->current_config) {
+		return USBD_REQ_NOTSUPP;
+	}
+
+	/* FIXME: assuming (usbd_dev->current_config - 1) offset
+	 *   gives the configuration descriptor "desc" which has
+	 *   desc->bConfigurationValue == usbd_dev->current_config */
+	cfx = &usbd_dev->config[usbd_dev->current_config - 1];
 
 	if (req->wIndex >= cfx->bNumInterfaces) {
 		return USBD_REQ_NOTSUPP;
 	}
 
 	*len = 1;
+	/* FIXME: assuming (req->wIndex) offset
+	 *   gives the interface descriptor "iface" which has
+	 *   iface->bInterfaceNumber == req->wIndex and
+	 *     iface->bAlternateSetting == 0 */
```

Comment:
right, but then we should document that the config's array provided in the exact format. (same for interface array).
most examples are following the norm but, we still need to document that.
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/usb/usbstd.h#L165
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/usb/usbd.h#L64
note: the descriptor is _provided_ by application code. and is "kept" inside.
i will remove all these FIXME if we put it somewhere to follow the norm in doc